### PR TITLE
[Table Improvements] Step 4 - Merge multiple cells 

### PR DESCRIPTION
**Pull request by juliaroldi**
Add support to merge multiple cells with cell selection. 
mergeCellsRooster

### CODE REVIEW

File: packages/roosterjs-editor-dom/lib/selection/Position.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default class Position implements NodePosition {
         offsetOrPosType: number = 0,
         private readonly isFromEndOfRange?: boolean
     ) {
-        if ((<NodePosition>nodeOrPosition).node) {
+        if (nodeOrPosition && (<NodePosition>nodeOrPosition).node) {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/selection/Position.ts:43]: This is interesting. In what case you see this is null? I'm afraid there maybe more serious bug hiding here

File: demo/scripts/controls/ribbonButtons/tableEditOperations.ts [hunk 2]
```diff
@@ -21,19 +24,25 @@ const tableEditOperationsLabel: Record<TableEditOperationsKey, string> = {
     insertBelow: 'Insert Below',
     insertLeft: 'Insert Left',
     insertRight: 'Insert Right',
+    merge: 'Merge Cells',
     deleteTable: 'Delete Table',
     deleteRow: 'Delete Row',
     deleteColumn: 'Delete Column',
+    mergeAbove: 'Merge Above',
+    mergeBelow: 'Merge Below',
```
Comment (JiuqingSong) [inline demo/scripts/controls/ribbonButtons/tableEditOperations.ts:32]: what about merge left and right?

File: packages/roosterjs-editor-dom/lib/table/VTable.ts [hunk 3]
```diff
@@ -348,6 +348,33 @@ export default class VTable {
                 }
                 break;
 
+            case TableOperation.MergeCells:
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/table/VTable.ts:351]: Question: is it possible to just reuse MergeLeft and MergeAbove to do MergeCells? e.g.
case MergeCells:
  for (...) for (...) this.edit(MergeLeft) ;

File: demo/scripts/controls/ribbonButtons/tableEditOperations.ts (hunk 2 above)
Comment (juliaroldi) [reply demo/scripts/controls/ribbonButtons/tableEditOperations.ts:32]: I just added this two to test the behavior, I will add the other two.

File: packages/roosterjs-editor-dom/lib/selection/Position.ts (hunk 1 above)
Comment (juliaroldi) [reply packages/roosterjs-editor-dom/lib/selection/Position.ts:43]: Sometimes, after just hovering the mouse while using cell selection after merging some cells, some node returned null.

Comment (JiuqingSong) [reply packages/roosterjs-editor-dom/lib/selection/Position.ts:43]: In that case I'd like to see why that can happen, and fix the root cause, but not just add a simple check here. We already enabled strict mode for this class and the in theory it should never be null.

Comment (juliaroldi) [reply packages/roosterjs-editor-dom/lib/selection/Position.ts:43]: I found the problem in selectTable, in this code 
`table.rows.item(coordinates.firstCell.y).cells.item(coordinates.firstCell.x),`
the the coordinates of the merged cells are being passes, so the it returns null to new Position.
After merge, we need to recalculate the coordinates.

Comment (juliaroldi) [reply packages/roosterjs-editor-dom/lib/selection/Position.ts]: I didn't create nothing to recalculate, just added a check to select a cell, if it exist.

File: packages/roosterjs-editor-dom/lib/table/VTable.ts (hunk 3 above)
Comment (juliaroldi) [reply packages/roosterjs-editor-dom/lib/table/VTable.ts:327]: I put the common code between the merges function inside a function

File: packages/roosterjs-editor-dom/lib/table/VTable.ts [hunk 4]
```diff
@@ -474,6 +465,21 @@ export default class VTable {
         }
     }
 
+    mergeCells(cell: VCell, nextCell: VCell, horizontally?: boolean) {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-dom/lib/table/VTable.ts:468]: make it private

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**